### PR TITLE
Simple Track Scheduler with Error handling

### PR DESCRIPTION
**Pull request by janeson332**

### CODE REVIEW

File: src/Utils/SimpleScheduler.cpp [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+  }
+
+  if (!currentOpt.has_value()) {
+    return false;
+  }
+
+  auto playingTrack = std::get<QueuedTrack>(playingTrackRet);
+
+  if (!currentOpt.value().isPlaying) {
+    mMusicBackend->play();
```
Comment (wurmmi) [inline src/Utils/SimpleScheduler.cpp]: why is this done here?
wouldn't this always override a user who just clicked pause?

File: src/Utils/SimpleScheduler.h [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+ private:
+  /**
+   * @brief Schedules one track after another.
+   * @details The next track is set to play, when the currently playing track
+   * reaches its' end. The thread only wakes up shortly before a tracks' end.
+   * The function also performs a consistency check between the DataStore and
+   * Spotify.
+   * @return True on success, false otherwise.
+   * Errors are printed within the function.
+   */
```
Comment (wurmmi) [inline src/Utils/SimpleScheduler.h]: adapt this comment; move it to the respective function doSchedule

File: src/Utils/SimpleScheduler.cpp [hunk 3]
```diff
@@ -0,0 +1,265 @@
+/*****************************************************************************/
+/**
+ * @file    TrackScheduler.cpp
+ * @author  Michael Wurm <wurm.michael95@gmail.com>
+ * @brief   Class TrackScheduler implementation
```
Comment (wurmmi) [inline src/Utils/SimpleScheduler.cpp]: adapt this comment; 
filename and author --> it is your implementation! :wink:

File: src/Utils/SimpleScheduler.cpp (hunk 1 above)
Comment (janeson332) [reply src/Utils/SimpleScheduler.cpp]: This function gets only called in the CheckPlaying state. Sure this funtion overrides the clicked pause, but this is only a short period when the track starts. With this I can now be sure if a track starts playing, is playing or finished

File: src/Utils/SimpleScheduler.h (hunk 2 above)
Comment (janeson332) [reply src/Utils/SimpleScheduler.h]: Haha yes this comments were copied from your implementation, and until now I was too lazy to adapt it :sweat_smile:

**Comment by janeson332** [comment]
please review, and check if skip track works correctly (issue #68). Also the inconsistency error on next song should be fixed (issue #59) - this was tested with about 20 tracks. (btw, during this test the automatic refresh of the access token also seemed to work)

File: src/Utils/SimpleScheduler.cpp [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+  // only check inconsistency when something is in the queue and state is in
+  // playing state
+  if (!empty && mSchedulerState == SchedulerState::Playing) {
+    return true;
+  }
+  return false;
+}
+
+/*
+TResultOpt SimpleScheduler::isDataIncosistent() {
```
Comment (wurmmi) [inline src/Utils/SimpleScheduler.cpp]: did you want to remove this function?

Comment (janeson332) [reply src/Utils/SimpleScheduler.cpp]: ups forgot its still there
deleted now

**Pull request merged**